drivers/overlay/encryption.go

```diff
@@ -143,17 +161,23 @@ func setupEncryption(localIP, advIP, remoteIP net.IP, vni uint32, em *encrMap, k
 		logrus.Warn(err)
 	}
 
+	var eel []*encrElem
+
 	for i, k := range keys {
-		spis := &spi{buildSPI(advIP, remoteIP, k.tag), buildSPI(remoteIP, advIP, k.tag)}
+		ee := &encrElem{
```

Comment:
I think what you need is a method that builds SPIs and Keys and returns the whole object, no? instead of computing them individually.